format/Message.fbs

```diff
@@ -99,6 +99,12 @@ table RecordBatch {
 
   /// Optional compression of the message body
   compression: BodyCompression;
+
+  /// Some types such as Utf8View are represented using a variable number of buffers.
+  /// For each such Field in the pre-ordered flattened logical schema, there will be
+  /// an entry in variadicCounts to indicate the number of extra buffers which belong
+  /// to that Field.
+  variadicCounts: [long];
```

Comment:
We probably need to bump the MetadataVersion to V6 to go along with this as anything without the `variadicCounts` field would not be able to safely read a Utf8View/BinaryView field and it should be documented that `<=V5` should reject those types.

Reply:
Unless I'm mistaken, this change is backwards compatible with V5. If a pre-Utf8View V5 reader encounters a schema including a field of Utf8View, I'd expect it to recognize that as a type it's unprepared to deal with and fail gracefully from there (before any record batches needed to be read correctly). For comparison, we didn't bump this for REE

Reply:
I'm mostly thinking that we should bump it because of the addition of the field to the RecordBatch table. But that's a good point. Most of the previous version changes (like from V4 -> V5) were also backwards compatible.